### PR TITLE
editoast: transform loading gauge type to enum

### LINKED ISSUE #6852: Adapt loading gauge of rolling stock model into an enum field
## Description

We can transform the `loading_gauge` field of `RollingStockModel` (model v2) to make it an enum field.

An example of enum field can be found here: [TrainScheduleV2](https://github.com/osrd-project/osrd/blob/f2d9e036aa09b1a08e2d28abb75e38e0ff04d30a/editoast/src/modelsv2/train_schedule.rs#L22-L23)

## What should be done

- [ ] Create a migration that transforms the table column into a `SMALLINT`
- [ ] Make `LoadingGaugeType` derive `strum_macros::FromRepr`
- [ ] Adapt the type of the field and use the annotation `#[model(to_enum)]`
- [ ] Adapt where this field was used

### PR DESCRIPTION
**Pull request by hamz2a**
closes #6852

### CODE REVIEW

File: editoast/migrations/2024-03-18-141345_change_loading_gauge_to_smallint/up.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        WHEN 'GB1' THEN 4
+        WHEN 'GC' THEN 5
+        WHEN 'FR3.3' THEN 6
+        WHEN 'FR3.3/GB/G2' THEN 7
+        WHEN 'GLOTT' THEN 8
+    END;
+
+ALTER TABLE rolling_stock DROP COLUMN loading_gauge;
+
+ALTER TABLE rolling_stock RENAME COLUMN loading_gauge_tmp TO loading_gauge;
```
Comment (younesschrifi) [inline editoast/migrations/2024-03-18-141345_change_loading_gauge_to_smallint/up.sql]:
```suggestion
UPDATE rolling_stock
SET loading_gauge = CASE
        loading_gauge
        WHEN 'G1' THEN 0
        WHEN 'G2' THEN 1
        WHEN 'GA' THEN 2
        WHEN 'GB' THEN 3
        WHEN 'GB1' THEN 4
        WHEN 'GC' THEN 5
        WHEN 'FR3.3' THEN 6
        WHEN 'FR3.3/GB/G2' THEN 7
        WHEN 'GLOTT' THEN 8
    END;
    
ALTER TABLE rolling_stock
ALTER COLUMN loading_gauge TYPE smallint USING(loading_gauge::smallint);
```

**Review by younesschrifi — COMMENTED** [review]
Nice PR ! 
Some changes left :)

File: editoast/migrations/2024-03-18-141345_change_loading_gauge_to_smallint/up.sql (hunk 1 above)
Comment (hamz2a) [reply editoast/migrations/2024-03-18-141345_change_loading_gauge_to_smallint/up.sql]: Done.

**Review by younesschrifi — APPROVED** [review]

**Review by woshilapin — APPROVED** [review]
Looks good to me. Didn't know about the `strum_macros::FromRepr` so I learned something 😄

**Pull request merged**